packages/blade/src/components/Dropdown/_decisions/controlled-dropdown.md

```diff
@@ -0,0 +1,342 @@
+
+```jsx
+const [currentSelection, setCurrentSelection] = React.useState();
+
+const onSomeAction = () => {
+  setCurrentSelection('mumbai');
+};
+
+const onResetClick = () => {
+  setCurrentSelection(undefined);
+};
+
+const onSelectChange = ({ values }) => {
+  setCurrentSelection(values[0]);
+};
+
+<Dropdown>
+  <SelectInput
+    label="City"
+    // New Prop 👇🏼
+    value={currentSelection}
+    onChange={onSelectChange}
+  />
+  <DropdownOverlay>
+    <ActionList>
+      <ActionListItem title="Mumbai" value="mumbai" />
+      <ActionListItem title="Bangalore" value="bangalore" />
+    </ActionList>
+  </DropdownOverlay>
+</Dropdown>;
+```
+
+</details>
+
+<details>
+<summary><code>useSelect</code> hook</summary>
+
+#### useSelect
+
+```jsx
+const { selectItem, reset, dropdownRef } = useSelect();
+
+const onSomeAction = () => {
+  selectItem('mumbai');
+};
+
+const onResetClick = () => {
+  reset();
+};
+
+<Dropdown ref={dropdownRef}>
+  <SelectInput />
+  <DropdownOverlay>
+    <ActionList>
+      <ActionListItem title="Mumbai" value="mumbai" />
+      <ActionListItem title="Bangalore" value="bangalore" />
+    </ActionList>
+  </DropdownOverlay>
+</Dropdown>;
+```
+
+Inspirations
+
+</details>
+
+## Concluded Solution
+
+Based on the [discussions](#approach-comparisons), we decided to go with a combination of approaches propsed above-
+
+- On `SelectInput`, `value` + `onChange` + `defaultValue`
+- On `ActionListItem`, `isSelected` + `onClick` + `isDefaultSelected`
+
+### When Trigger is `SelectInput`
+
+For SelectInput (and maybe AutoComplete in future), we decided to go with `value` + `onChange` props on the trigger. This keeps the controlled state props on the same component as opposed to `isSelected` + `onChange` approach where `isSelected` goes on ActionListItem and `onChange` goes on SelectInput.
+
+```jsx
+const [currentSelection, setCurrentSelection] = React.useState();
+
+const onSomeAction = () => {
+  setCurrentSelection('mumbai');
+};
+
+const onResetClick = () => {
+  setCurrentSelection(undefined);
+};
+
+const onSelectChange = ({ values }) => {
+  setCurrentSelection(values[0]);
+};
+
+<Dropdown>
+  <SelectInput
+    label="City"
+    // New Prop 👇🏼
+    value={currentSelection}
+    onChange={onSelectChange}
+  />
+  <DropdownOverlay>
+    <ActionList>
+      <ActionListItem title="Mumbai" value="mumbai" />
+      <ActionListItem title="Bangalore" value="bangalore" />
+    </ActionList>
+  </DropdownOverlay>
+</Dropdown>;
+```
+
+### When Trigger is Button or other triggers
+
+When it's Button or other trigger, we use `isSelected` and `onClick` from `ActionListItem`
+
+```jsx
+const [currentSelection, setCurrentSelection] = React.useState();
+const [isDropdownOpen, setIsDropdownOpen] = React.useState(false);
+
+const onSomeAction = () => {
+  setCurrentSelection('mumbai');
+};
+
+<Dropdown>
+  <DropdownButton>Show Dropdown</DropdownButton>
+  <DropdownOverlay>
+    <ActionList>
+      <ActionListItem
```

Comment:
When we have dropdown button, wouldn't we need to update the text of the button to the selected item? 🤔

Reply:
nope. That should ideally be a SelectInput if you have that usecase. Although if someone wants to have that, they can still do that since anyway the selection is controlled here. Won't recommend though. Most of the times, SelectInput would be the right component for such usecases.

Reply:
see this is a new transactions module
<img width="863" alt="Screenshot 2023-05-08 at 5 23 23 PM" src="https://user-images.githubusercontent.com/11384858/236817059-5760237e-2499-4fd5-9c87-7f32aac69e09.png">